src/cascadia/TerminalApp/WslDistroGenerator.cpp

```diff
@@ -97,6 +99,13 @@ std::vector<TerminalApp::Profile> WslDistroGenerator::GenerateProfiles()
         {
             std::wstring distName;
             std::getline(wlinestream, distName, L'\r');
+
+            if (distName.substr(0, std::min(distName.size(), DockerDistributionPrefix.size())) == DockerDistributionPrefix)
+            {
+                // Skip all of the known Docker WSL2 distributions.

```

Comment:
Could we maybe elaborate more, and link back to the github thread with more details? This isn't immediately apparent why you'd want to skip these

Reply:
```suggestion
                // Docker for Windows creates some utility distributions to handle Docker commands.
                // Pursuant to GH#3556, because they are _not_ user-facing we want to hide them.
```